parser.go

```diff
@@ -166,16 +166,18 @@ func (p *ParseContext) Next() *Token {
 		return &Token{Index: p.argi, Type: TokenEOL}
 	}
 
+	if p.argi > 0 && p.argi <= len(p.rawArgs) && p.rawArgs[p.argi-1] == "--" {
```

Comment:
It's not clear to me why this works, but the below doesn't?

Reply:
I must admit that I don't fully own the parsing logic, so correct me if I'm wrong. But my understanding is the following:

If `my-app --` is parsed, the last part is interpreted as a `--` delimiter and `argsOnly` is set to `true`. For normal command parsing, this is OK. But for autocompletion, this means that the last `--` is not treated as the start of a flag. 

By moving setting `argsOnly = true` to the next call of `Next()`, a trailing `--` does not cause `argsOnly` to be set to `true`.

If there is a more elegant solution to access the previous argument, I'm all ears.